### ISSUE TITLE
Migrate api-documentation-tests to use pytest and pytest-xdist to parallelize tests execution

### ISSUE THREAD

**Issue by amol-** [rstudio/rsconnect#1124]
`api-documentation-tests` is one of the CI steps taking the longest to complete.
They are implemented using `nose` and they run single threaded, as the GitHub runners have 4 cores, there might be a benefit we can get from parallelizing the tests.

There is no guarantee they can be parallelized (as they might depend on global shared states) and there is no guarantee that the improvement will be significant as they depend on an underlying connect instance. But it makes sense to check as a significant speed up would improve delivery time for everyone.

**Comment by aronatkins** [comment]
The API documentation tests start/stop Connect repeatedly. Each test assumes that it owns that Connect instance.

We want to parallelize execution of these tests, but we'll need to do more than shifting how the Python tests run.

**Comment by amol-** [comment]
> The API documentation tests start/stop Connect repeatedly. Each test assumes that it owns that Connect instance.
> 

Yes, I had the impression that was happening, that's why I was hypothetical, it can turn out to be something easy or something super complex, I'd have to look into it in more detail to make an estimate. But I think it makes sense to at least investigate how hard it would be.
